redis-sink/src/main/java/org/springframework/cloud/stream/module/redis/CustomRedisConfiguration.java

```diff
@@ -0,0 +1,64 @@
+package org.springframework.cloud.stream.module.redis;
+
+import java.net.UnknownHostException;
+
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import redis.clients.jedis.Jedis;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingClass;
+import org.springframework.boot.autoconfigure.redis.RedisAutoConfiguration;
+import org.springframework.boot.autoconfigure.redis.RedisProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.redis.connection.jedis.JedisConnection;
+import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
+import org.springframework.data.redis.core.RedisOperations;
+
+/**
+ * Creates a dedicated RedisConnectionFactory (that is, different from the spring-cloud-stream one).
+ *
+ * @author Eric Bottard
+ */
+@Configuration
+@ConditionalOnClass({ JedisConnection.class, RedisOperations.class, Jedis.class })
+@EnableConfigurationProperties
+public class CustomRedisConfiguration extends RedisAutoConfiguration {
```

Comment:
It isn't registered as such in spring.factories.
Le 5 août 2015 21:47, "Mark Pollack" <notifications@github.com> a écrit :

> In
> redis-sink/src/main/java/org/springframework/cloud/stream/module/redis/CustomRedisConfiguration.java
> <https://github.com/spring-cloud/spring-cloud-stream-modules/pull/1#discussion_r36345736>
> :
>
> > +import org.springframework.boot.context.properties.EnableConfigurationProperties;
> > +import org.springframework.context.annotation.Bean;
> > +import org.springframework.context.annotation.Configuration;
> > +import org.springframework.data.redis.connection.jedis.JedisConnection;
> > +import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
> > +import org.springframework.data.redis.core.RedisOperations;
> > +
> > +/**
> > + * Creates a dedicated RedisConnectionFactory (that is, different from the spring-cloud-stream one).
> > + *
> > + * @author Eric Bottard
> > + */
> > +@Configuration
> > +@ConditionalOnClass({ JedisConnection.class, RedisOperations.class, Jedis.class })
> > +@EnableConfigurationProperties
> > +public class CustomRedisConfiguration extends RedisAutoConfiguration {
>
> It is an Autoconfiguration, I don't understand.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spring-cloud/spring-cloud-stream-modules/pull/1/files#r36345736>
> .
>